app/models/normalize_worldcat.rb

```diff
@@ -0,0 +1,37 @@
+# Tranforms results from {SearchWorldcat} into normalized {Result}s
+#
+class NormalizeWorldcat
+  # Translate Google results into local result model
+  def to_result(results)
+    norm = {}
+    xml = Nokogiri::XML.parse(results.body)
+    norm['total'] = xml.xpath('//xmlns:numberOfRecords').text.to_i
+    norm['results'] = []
+    extract_results(xml, norm)
+    norm
+  end
+
+  private
+
+  # Extract the information we care about from the raw results and add them
+  # return them as an array of {result}s
+  def extract_results(xml, norm)
+    xml.xpath('//xmlns:oclcdcs').each do |item|
+      result = Result.new(item.xpath('dc:title').text.strip,
+                          item.xpath('dc:date').text.strip,
+                          'http://www.worldcat.org/oclc/' + item.xpath('oclcterms:recordIdentifier[not(@xsi:type)]').text.strip,
```

Comment:
I'd suggest extracting this link creating logic to it's own method to make this easier to read:

```
  def item_link(item)
    'http://www.worldcat.org/oclc/' +
      item.xpath('oclcterms:recordIdentifier[not(@xsi:type)]').text.strip
  end
```